### PR TITLE
FluidSynth configuration improvements

### PR DESCRIPTION
**Pull request by johnnovak**
# Description

As per the release notes, plus some lightweight refactoring/cleanup and a substantial improvement of the FluidSynth-related config setting descriptions.

This PR also showcases my long-term vision for our documentation (which currently only consists of the Getting Started guide): code changes and documentation changes are made _in lockstep_, in the _same_ PR!

This was one of the main reasons for keeping the doco and the sources in the same repo.

## Related issues

Specifying a user-defined SoundFont directory was brought up by @AnonTet; see this discussion:
- https://github.com/dosbox-staging/dosbox-staging/pull/3981#issuecomment-2425260254

# Release notes

- Support for the second volume scaling parameter in the `soundfont` setting has been removed; use the new `soundfont_volume` setting for SoundFont-level volume adjustmens.

- You can now specify a user-defined SoundFont directory via the new `soundfont_dir` setting. When defined, DOSBox Staging attempts to load the SoundFont from this directory first before trying the other standard
  locations.

- SoundFonts are now sorted by their filenames in the `mixer /listmidi` command's output.

- Changing any setting from the `[fluidsynth]` section (e.g., `fsynth_reverb`, `soundfont_dir`) will take effect immediately and will trigger the reloading of the current SoundFont if FluidSynth is active.

# Documentation update

image

# Manual testing

- Tested the new config settings at runtime, making sure the changes take effect as described in the notes.

- Here are the new/revised setting descriptions:

image

image

image

# Checklist

_Please tick the items as you have addressed them. Don't remove items; leave the ones that are not applicable unchecked._

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [x] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [x] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [x] provided the release notes draft (for significant user-facing changes).
- [x] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [x] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

**Comment by johnnovak** [comment]
@Python-Exoproject I don't think you'll be affected by the `soundfont_volume` change or any of the other changes, but I tagged you, just in case.

File: src/midi/midi_fluidsynth.cpp [hunk 1]
```diff
@@ -192,15 +175,40 @@ static std::deque<std_fs::path> get_data_dirs()
...
+	auto dirs = get_platform_data_dirs();
+
+	auto sf_dir = get_fluidsynth_section()->Get_string("soundfont_dir");
+	if (!sf_dir.empty()) {
+		// The user-provided SoundFont dir might use a different casing
+		// of the actual path on Linux & Windows, so we need to
+		// normalise that to avoid some subtle bugs downstream (see
+		// `find_sf_file()` as well).
+		dirs.push_back(std_fs::canonical(sf_dir));
+	}
```
Comment (interloper98) [inline src/midi/midi_fluidsynth.cpp:199]: Maybe give the user's directory first billing by putting it first ` dirs.insert(dirs.begin(), ...)`

**Review by interloper98 — APPROVED** [review]
Just one minor suggestion from me. Documentation improvements look good!

File: src/midi/midi_fluidsynth.cpp (hunk 1 above)
Comment (johnnovak) [reply src/midi/midi_fluidsynth.cpp:199]: Ah, that's my mistake. That's what I _wanted_ to do and that's what I said in the release notes. Good catch! 🎖️

Comment (johnnovak) [reply src/midi/midi_fluidsynth.cpp]: Done.

**Comment by johnnovak** [comment]
> Just one minor suggestion from me. Documentation improvements look good!

Thanks man! 😎

**Comment by AnonTet** [comment]
New fluidsynth options work as expected. soundfont_dir and separating the volume to it's own option is more intuitive if you ask me.
Also, the new doc helps a lot.
Thanks.

**Comment by johnnovak** [comment]
> New fluidsynth options work as expected. soundfont_dir and separating the volume to it's own option is more intuitive if you ask me. Also, the new doc helps a lot. Thanks.

Thanks for the feedback @AnonTet. I'm glad to hear you enjoy the FluidSynth enhancements 👍🏻

**Pull request merged**